display_books_promise.js

```diff
@@ -0,0 +1,66 @@
+let BookData = [];
+
+const insertDataInHtml = (data) => {
+  data.sort((a, b) => a.price - b.price);
+  const tableBody = document.getElementById('table-all-books');
+  tableBody.innerHTML = '';
+  const headerRow = document.createElement('tr');
+  headerRow.innerHTML = `
+    <th>BookId</th>
+    <th>Genre</th>
+    <th>Price</th>
+    <th>Examine</th>
+   `;
+  tableBody.appendChild(headerRow);
+  data.forEach((book) => {
+    const row = document.createElement('tr');
+    row.innerHTML = `
+       <th>${book.bookId}</th>
+       <th>${book.genre}</th>
+       <th>${book.price}</th>
+      `;
+    tableBody.appendChild(row);
+  });
+};
+
+fetch('random_books.json')
+  .then((response) => response.json())
+  .then((data) => {
+    BookData = data;
+    insertDataInHtml(BookData);
```

Comment:
What is the sense of assigning the data to BookData as you are directly using that data in insertDataInHtml. 

```
insertDataInHtml(data);
```